### PR TITLE
Test and implement proper parenthetics

### PR DESCRIPTION
**Pull request by welliam**

### CODE REVIEW

File: src/proper_parens.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+from __future__ import unicode_literals
+
+
+class Stack(list):
+    # so the code below works
+    push = list.append
```
Comment (cewing) [inline src/proper_parens.py]: nice minimal stack :)

File: src/proper_parens.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    stack = Stack()
+    for c in s:
+        if c == '(':
+            stack.push(';^3')
+        elif c == ')':
+            try:
+                stack.pop()
+            except IndexError:
+                return -1
+    return min(len(stack), 1)
```
Comment (cewing) [inline src/proper_parens.py]: not exactly.  The assignment says there are three valid return values.  -1, 0 or 1.  How many does your implementation have?

File: src/proper_parens.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    opens = 0
+    for c in s:
+        if c == '(':
+            opens += 1
+        elif c == ')':
+            if opens:
+                opens -= 1
+            else:
+                return -1
+    return min(opens, 1)
```
Comment (cewing) [inline src/proper_parens.py]: same problem exists here.

**Comment by welliam** [comment]
If opens is 0, line 34 will return 0. Otherwise, it will return 1.  Line 33 is the third value, -1.

**Pull request merged**